cmake/Modules/FindAVX.cmake

```diff
@@ -12,6 +12,23 @@ SET(AVX_CODE "
   }
 ")
 
+SET(AVX512_CODE "
+  #include <immintrin.h>
+
+  int main()
+  {
+    __m512i a = _mm512_set_epi8(0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0);
```

Comment:
Does this actually test all of the AVX512 subsets you're turning on? (-mavx512f -mavx512dq -mavx512vl -mavx512bw)

Reply:
No, there's one old Intel  on which this test will be a false-positive.
I'll add another intrinsic to make it fool-proof.

Reply:
Intel KNC doesn't support AVX512BW but supports AVX512VL & AVX512DQ.
So I'll also include an AVX512BW intrinsic.
Thank you!